src/search/search-index-new/search/filters.ts

```diff
@@ -86,17 +106,35 @@ const incDomainSearch = ({ domains }: Partial<SearchParams>) =>
 const excDomainSearch = ({ domainsExclude }: Partial<SearchParams>) =>
     domainSearch(domainsExclude)
 
+
+async function listSearch({ lists }: Partial<SearchParams>) {
+    if (!lists || !lists.length) {
+        return undefined
+    }
+    const customList = new CustomListBackground({ storageManager })
+
+    const urls = new Set<string>()
+
+    // The list filter contains only one list at a time
+    // It is just a temporary hack until multiple lists for filtering in used.
+    const listEnteries = await customList.fetchListPages(Number(lists[0]))
+    listEnteries.map(({ pageUrl }: any) => urls.add(pageUrl))
```

Comment:
You generally only want to use `Array.prototype.map` to iterate over an array, performing some change to each element, then returning a new array made up of those changed elements. `Array.prototype.forEach` would seem more appropriate here. Or you could keep using `map`, but use the return value to construct your `urls` Set. i.e,

```ts
const urls = new Set<string>(listEntries.map(entry => entry.pageUrl as string))
```

(tbh not sure if that `as string` type assertion is needed off the top of my head) 

Also minor typo in `listEnteries`